engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -2116,9 +2126,25 @@ namespace dmGameSystem
 
     dmGameObject::PropertyResult CompGuiSetProperty(const dmGameObject::ComponentSetPropertyParams& params) {
         GuiComponent* gui_component = (GuiComponent*)*params.m_UserData;
-        if (params.m_PropertyId == PROP_MATERIAL) {
+        dmhash_t set_property = params.m_PropertyId;
+        if (set_property == PROP_MATERIAL) {
             return SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, MATERIAL_EXT_HASH, (void**)&gui_component->m_Material);
         }
+        else if (set_property == PROP_FONTS) {
+            if (!params.m_Options.m_Key) {
+                return dmGameObject::PROPERTY_RESULT_INVALID_KEY;
+            }
+            dmRender::HFontMap font = 0x0;
+            dmGameObject::PropertyResult res = SetResourceProperty(dmGameObject::GetFactory(params.m_Instance), params.m_Value, FONT_EXT_HASH, (void**)&font);
+            dmGui::HScene scene = gui_component->m_Scene;
+            dmGui::Result r = dmGui::AddFont(scene, params.m_Options.m_Key, (void*) font, params.m_Value.m_Hash);
```

Comment:
How do we want to deal with "resetting" this property?
E.g. we have "sprite.reset_constant()" in order to go back to the previous version

Reply:
I think it's a good "nice to have" feature we would like to design and implement in the future because we wanna use it not only here, but as replacement for deprecated `sprite.reset_constant()` function.

At this moment, in this particular case with fonts and textures, it's easy to workaround by getting value before set, and then set it later again (reset).

Reply:
```
	local system_font = go.get("#gui", "fonts", {key = "system_font"})
	go.set("#gui", "fonts", self.font, {key = "system_font"})
	--reset
	go.set("#gui", "fonts", system_font, {key = "system_font"})
```